README.md

```diff
@@ -91,97 +106,181 @@ cloning over SSH may provide a better experience (which requires
     git clone git@github.com:apple/swift.git
     ./swift/utils/update-checkout --clone-with-ssh
 
-#### CMake
-[CMake](https://cmake.org) is the core infrastructure used to configure builds of 
-Swift and its companion projects; at least version 3.4.3 is required. 
+### Building Swift
 
-On macOS, you can download the [CMake Binary Distribution](https://cmake.org/download),
-bundled as an application, copy it to `/Applications`, and add the embedded
-command line tools to your `PATH`:
+The `build-script` is a high-level build automation script that supports basic
+options such as building a Swift-compatible LLDB, building the Swift Package
+Manager, building for iOS, running tests after builds, and more. It also
+supports presets, which you can define for common combinations of build options.
 
-    export PATH=/Applications/CMake.app/Contents/bin:$PATH
-    
-On Linux, if you have not already installed Swift's [development
-dependencies](#linux-dependencies), you can download and install the CMake
-package separately using the following command:
+View the inline help to find out more, in particular the section "Typical uses":
 
-    sudo apt-get install cmake
-    
+    utils/build-script -h
 
-#### Ninja
-[Ninja](https://ninja-build.org) is the current recommended build system
-for building Swift and is the default configuration generated by CMake. [Pre-built
-packages](https://github.com/ninja-build/ninja/wiki/Pre-built-Ninja-packages)
-are available for macOS and Linux distributions. You can also clone Ninja
-next to the other projects and it will be bootstrapped automatically:
+A basic command to build Swift with optimizations and run basic tests with
+Ninja:
 
-**Via HTTPS**
+    utils/build-script --release --test
```

Comment:
I strongly recommend building with assertions for both compiler and stdlib. Only disable for benchmarking and perf tuning (or release validation, which is out of scope for most contributors).